### PR TITLE
build: Add default Makefile command `help`

### LINKED ISSUE #3277: Improve Makefile behavior
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/issues/3354

Originally created by @pandemicsyn on 2022-03-29 03:16:59

---

Our Makefile doesn't have a help option AND defaults to executing a build if invoked without arguments. Makefile help/docs can be generated if you run `make explain_makefile` but you have to know that exists (or read the Makefile, in which case you no longer need the explainer). 
With some very minor changes (essentially this bit from https://gitlab.com/pandemicsyn/llamalake/-/blob/main/Makefile#L7 + a few inline comments), we could provide a help option, and use that as the default target:

```
(melty-3.8) ➜ make
api                            build api
base_image                     builds meltano/base
build                          build the ui and api
bundle                         clean up static assets and rerun UI build
clean                          clean statically generated assets
docker_images                  build docker base_image and prod_image
docker_sdist
docs_image
explain_makefile               launch a web server further explaining this makefile.
freeze_db
lint                           run python and eslint tasks
lint_eslint                    run eslint
lint_python                    run json/yaml validation and execute tox to perform python lint tasks
lock                           run poetry lock
prod_image                     builds meltano/meltano the AIO production image that includes static UI artifacts
release                        perform a release, should only be performed as part of the CI pipeline
show_lint                      show python and eslint ralated commands
test                           run tests
ui                             build the UI
```

And explicitly support `help` arg via

```

(melty-3.8) ➜ make help
api                            build api
base_image                     builds meltano/base
build                          build the ui and api
bundle                         clean up static assets and rerun UI build
clean                          clean statically generated assets
docker_images                  build docker base_image and prod_image
docker_sdist
docs_image
explain_makefile               launch a web server further explaining this makefile.
freeze_db
lint                           run python and eslint tasks
lint_eslint                    run eslint
lint_python                    run json/yaml validation and execute tox to perform python lint tasks
lock                           run poetry lock
prod_image                     builds meltano/meltano the AIO production image that includes static UI artifacts
release                        perform a release, should only be performed as part of the CI pipeline
show_lint                      show python and eslint ralated commands
test                           run tests
ui                             build the UI
```

### PR DESCRIPTION
**Pull request by WillDaSilva**
Can be invoked by running `make` or `make help`:

```
(meltano37) will ~/meltano/meltano (make-help) $ make
api              Build the Meltano API
base_image       Build the Meltano base image
build            Build the Meltano UI and API
bundle           Clean build artifacts, then build & bundle the UI
clean            Delete build artifacts
docker_images    Build the Meltano Docker images
docker_sdist     Build an image off of the base image that includes the Meltano sdist
explain          Bring up a web server with this Makefile annotated
freeze_db        Freeze the Meltano database to support DB migration during upgrades
help             Display this help text
lint_eslint      Run eslint & automatically apply fixes where possible
lint             Run linters & automatically apply fixes where possible
lint_python      Run Python linters & automatically apply fixes where possible
lock             Update Poetry lock file
prod_image       Build the Meltano prod image
release          Execute the automated steps of the deprecated release process
sdist            Build the Meltano sdist for development
sdist_public     Build the Meltano sdist for release
show_lint_eslint Run eslint & display output
show_lint_python Run Python linters & display output
show_lint        Run linters & display output
test             Run the tests
ui               Build the Meltano UI
```

I've removed `docs_image`, `docs_shell`, `docs/build`, and `docs/serve`, as none of them seem to work.

Going forward I think we ought to reduce the usage/size/importance of the Makefile, and ideally eventually get rid of it altogether. If need be we can use the `scripts` directory for scripts that need to be executed by devs. Dead code is tech debt that should be cleaned up when recognized, and I get the impression that this Makefile has a good amount of it. Anything in it that is only there for GitHub CI should be moved into the `.github` directory.

A reliance on Makefiles and arbitrary build/dev scripts encourages the addition of more of them, which in my experience tends to result in messes that are only ever understood by the original authors, if even. Keeping things constrained to a few commands using well-known tools (e.g. `poetry`, `yarn`, `pre-commit`, `pytest`, etc.), and then putting the rest into CI limits us in some ways, but can make for a cleaner project that is easier for newcomers to contribute to.

Nice work on [the implementation of the `help` command](https://gitlab.com/pandemicsyn/llamalake/-/blob/main/Makefile#L7) @pandemicsyn!

Closes #3277

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -176,36 +156,36 @@ JSON_YML_VALIDATE = poetry run python src/meltano/core/utils/validate_json_schem
...
 # Makefile Related Tasks
 # ======================
 #
-# - `make explain_makefile` will bring up a web server with this makefile annotated.
-explain_makefile:
-	docker stop explain_makefile || echo 'booting server'
-	${DOCKER_RUN} --name explain_makefile -p 8081:8081 node ./Makefile_explain.sh
+# - `make explain` will bring up a web server with this makefile annotated.
+explain: ## Bring up a web server with this Makefile annotated
+	docker stop explain || echo 'booting server'
```
Comment (pandemicsyn) [inline Makefile:180]: IMHO we could nuke `make explain` and the associated bash script as part of this PR.

**Review by pandemicsyn — APPROVED** [review]
lgmt.

**Comment by pandemicsyn** [comment]
fyi - added @alexmarple as a reviewer incase he uses some of the frontend specific make targets.

**Review by edgarrmondragon — APPROVED** [review]
👌

**Review by alexmarple — APPROVED** [review]
LGTM

**Pull request merged**